gsuite/resource_group_members.go

```diff
@@ -0,0 +1,249 @@
+	return &schema.Resource{
+		Create: resourceGroupMembersCreate,
+		Read:   resourceGroupMembersRead,
+		Update: resourceGroupMembersUpdate,
+		Delete: resourceGroupMembersDelete,
+
+		Schema: map[string]*schema.Schema{
+			"group": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"member": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: schemaMember,
+				},
+			},
+		},
+	}
+}
+
+func resourceGroupMembersRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	members, err := getApiMembers(d.Id(), config)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("group", d.Id())
+	d.Set("member", membersToCfg(members))
+	return nil
+}
+
+func resourceGroupMembersCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG]: Creating gsuite_group_members")
+	gid, err := createOrUpdateGroupMembers(d, meta)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(gid)
+	return resourceGroupMembersRead(d, meta)
+}
+
+func resourceGroupMembersUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG]: Updating gsuite_group_members")
+	_, err := createOrUpdateGroupMembers(d, meta)
+
+	if err != nil {
+		return err
+	}
+	return resourceGroupMembersRead(d, meta)
+}
+
+func resourceGroupMembersDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG]: Deleting gsuite_group_members")
+	config := meta.(*Config)
+
+	for _, raw_member := range d.Get("member").(*schema.Set).List() {
+		member := raw_member.(map[string]interface{})
+		deleteMember(member["email"].(string), d.Id(), config)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func membersToCfg(members *directory.Members) []map[string]interface{} {
+	if members == nil {
+		return nil
+	}
+
+	finalMembers := make([]map[string]interface{}, 0, len(members.Members))
+
+	for _, m := range members.Members {
+		finalMembers = append(finalMembers, map[string]interface{}{
+			"email":  m.Email,
+			"etag":   m.Etag,
+			"kind":   m.Kind,
+			"status": m.Status,
+			"type":   m.Type,
+			"role":   m.Role,
+		})
+	}
+
+	return finalMembers
+}
+
+func resourceMembers(d *schema.ResourceData) (members []map[string]interface{}) {
+	for _, raw_member := range d.Get("member").(*schema.Set).List() {
+		member := raw_member.(map[string]interface{})
+		members = append(members, member)
+	}
+	return members
+}
+
+func createOrUpdateGroupMembers(d *schema.ResourceData, meta interface{}) (string, error) {
+	config := meta.(*Config)
+	gid := d.Get("group").(string)
+
+	// Get members from config
+	cfgMembers := resourceMembers(d)
+
+	// Get members from API
+	apiMembers, err := getApiMembers(gid, config)
+	if err != nil {
+		return gid, fmt.Errorf("Error updating memberships: %v", err)
+	}
+	// This call removes any members that aren't defined in cfgMembers,
+	// and adds all of those that are
+	err = reconcileMembers(d, cfgMembers, membersToCfg(apiMembers), config, gid)
```

Comment:
`gsuite/resource_group_members.go:127:52: cannot use apiMembers (type []*admin.Member) as type *admin.Members in argument to membersToCfg`